procfile/README.md

```diff
@@ -42,3 +42,13 @@ Specifies the command that should be run when executing this process
 ```yaml
 command: ./bin/web
 ```
+
+**Expose**
+
+Specifies the exposure settings for the process. This allows you to expose a tcp/http/https/ssl service on any process in the application.
+
+```yaml
+expose:
+  type: tcp // Possible options are tcp/http/https/ssl
```

Comment:
One scenario that I haven't really thought through with this yet is when you want to have multiple listeners. For example, our existing behavior when creating https load balancers is to create both an https and an http listener. Maybe we should do something like this:

```yaml
web:
  visibility: public
  ports:
    - "80:8080":
       protocol: http
    - "443:8080":
       protocol: https
    - "1234:8081":
       protocol: tcp
    - "4321:8081":
       protocol: ssl
```

This might be nice because it offers pretty close parity with `ports` in [docker-compose.yml](https://docs.docker.com/compose/compose-file/#ports).

The awkward bit is how we communicate the protocol, which we need to create the correct listeners on the load balancer. We could provide some sensible defaults, like:

```yaml
web:
  visibility: public
  ports:
    - 80:80 # http listener
    - 443:80 # https listener
    - "1234:1234" # tcp listener
    - # ??? not really a good way to handle an ssl listener.
```

This would also mean we could avoid setting the PORT environment variable and just let containers bind to the port they specify.